libs/beacon/include/beacon/trusted_dealer.hpp

```diff
@@ -0,0 +1,62 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "crypto/mcl_dkg.hpp"
+
+namespace fetch {
+namespace beacon {
+
+class TrustedDealer
+{
+  using MuddleAddress = byte_array::ConstByteArray;
```

Comment:
Style: public: first, then private: after usually.

Looking at this, couldn't it use unordered_map<MuddleAddress, DkgOutput> without the index, instead of DKGOutput do a return {};?

I'd have to check but this pattern of returning a const ref to something within this class can often be dangerous for memory safety - is it that the MCL stuff doesn't do copy constructors well (something I have painfully noticed)?